### PR TITLE
fix: invalidate cached subsidy request configuration after toggle

### PR DESCRIPTION
**Pull request by long74100**
- invalidate cache after updating subsidy request configuration
- relies on fix in https://github.com/openedx/frontend-platform/pull/311
- fix test that was using waitFor without await

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/SubsidyRequestManagementTable/tests/CourseTitleCell.test.jsx [hunk 1]
```diff
@@ -46,10 +46,9 @@ describe('CourseTitleCell', () => {
     renderWithRouter(Component);
     userEvent.click(screen.getByText(defaultProps.row.original.courseTitle));
     expect(screen.getByText('Loading course details...'));
-    waitFor(() => {
+    await waitFor(() => {
```
Comment (adamstankiewicz) [inline src/components/SubsidyRequestManagementTable/tests/CourseTitleCell.test.jsx:49]: 🎉 Good catch.

File: src/components/SubsidyRequestManagementTable/tests/CourseTitleCell.test.jsx [hunk 2]
```diff
@@ -46,10 +46,9 @@ describe('CourseTitleCell', () => {
...
       screen.getByText(mockCourseDetails.shortDescription);
       screen.getByText('Learn more about this course');
-      screen.getByLabelText('in a new tab');
```
Comment (adamstankiewicz) [inline src/components/SubsidyRequestManagementTable/tests/CourseTitleCell.test.jsx:52]: Why remove this line? It's here to ensure this links is in fact opening in a new tab.

File: src/components/subsidy-request-configuration/data/hooks.js [hunk 3]
```diff
@@ -18,59 +18,68 @@ export const useSubsidyRequestConfiguration = (enterpriseUUID) => {
...
+    if (!enterpriseUUID) {
+      return;
+    }
+
+    const loadConfiguration = async () => {
       try {
-        const response = await EnterpriseAccessApiService.getSubsidyRequestConfiguration(enterpriseUUID);
-        const config = camelCaseObject(response.data);
-        setSubsidyRequestConfiguration(config);
+        await loadSubsidyRequestConfiguration({});
```
Comment (adamstankiewicz) [inline src/components/subsidy-request-configuration/data/hooks.js:77]: nit: could the function argument default to `{}` if not provided so, `{}` isn't need here?

File: src/components/SubsidyRequestManagementTable/tests/CourseTitleCell.test.jsx (hunk 2 above)
Comment (long74100) [reply src/components/SubsidyRequestManagementTable/tests/CourseTitleCell.test.jsx:52]: Ah it's part of hyperlink. `getByText` will work here, not `getByLabelText`

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**